pkg/render/intrusion_detection.go

```diff
@@ -273,39 +281,55 @@ func (c *intrusionDetectionComponent) intrusionDetectionJobServiceAccount() *v1.
 }
 
 func (c *intrusionDetectionComponent) intrusionDetectionClusterRole() *rbacv1.ClusterRole {
-	return &rbacv1.ClusterRole{
-		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "intrusion-detection-controller",
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{
+				"projectcalico.org",
+			},
+			Resources: []string{
+				"globalalerts",
+				"globalalerts/status",
+				"globalthreatfeeds",
+				"globalthreatfeeds/status",
+				"globalnetworksets",
+			},
+			Verbs: []string{
+				"*",
+			},
 		},
-		Rules: []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{
+				"crd.projectcalico.org",
+			},
+			Resources: []string{
+				"licensekeys",
+			},
+			Verbs: []string{
+				"get", "watch",
+			},
+		},
+	}
+	if !c.managedCluster {
```

Comment:
Just out of curiosity, do we still need this role at all in managed clusters?

Reply:
Right, IDS controller in managed cluster need permission to access licensekeys, globalthreatfeeds and globalnetworksets. 
I'll remove the permissions for globalalerts, they are not needed in managed cluster anymore.

Reply:
I tested by removing globalalerts permission for IDS controller in managed cluster, if we do that then IDS in management doesn't have access to globalalerts in managed cluster. So we need to retain those permissions.